### PR TITLE
add NUI Components guide

### PR DESCRIPTION
**Pull request by junqingma**
add guide for Tizen.NUI.Components

### CODE REVIEW

**Review by elishateng — APPROVED** [review]
Look good to me.

**Review by huayongxu — APPROVED** [review]
It is great!

**Review by liangliangzh — APPROVED** [review]
it looks good to me

File: docs/nui-components/Guide.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Welcome to CommonUI
```
Comment (agnelovaz) [inline docs/nui-components/Guide.md:1]: We are using the name NUI Components instead of CommonUI I believe.

File: docs/nui-components/Guide.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+This guide will take you through what you need to know to produce efficient, dynamic and visually rich applications in CommonUI easily
+
+## What is CommonUI
+Cover the common UI components of FHUB, TV, Mobile, Watch.
```
Comment (agnelovaz) [inline docs/nui-components/Guide.md:5]: No need to mention targets as want NUI to seem general and useful for multiple applications.

File: docs/nui-components/Guide.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Support dynamic resize and re-layout
+Support theme
+Support style
+Support R2L
```
Comment (agnelovaz) [inline docs/nui-components/Guide.md:9]: RtL but shouldn't use abbreviations like that, "Support right to left languages"

File: docs/nui-components/Guide.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+Support multi-resolution
+Support various input method (remote control, touch, gesture…)
+Support global focus system (animation & auto-navigation)
+
+## Getting Started
+CommonUI repositories are avaiable in GitHub with instructions to install and build.
```
Comment (agnelovaz) [inline docs/nui-components/Guide.md:15]: link to the github repo or remove this instruction

File: docs/nui-components/Button.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+![ButtonProperty](../../assets/images/components/ButtonProperty.PNG)
+
+Button created by property pressed:
+
+![ButtonPropertyPressed](../../assets/images/components/ButtonPropertyPressed.PNG)
+## Create with attributes
+1. Create button attributes
+
+~~~{.cs}
+ButtonAttributes familyBasicButtonAttributes = new ButtonAttributes
```
Comment (agnelovaz) [inline docs/nui-components/Button.md:60]: Its nice to have a detailed example of a button with many levels of images and 2 styles but first we should start with a simple button example.  A single background image and some text.  Later in the guide we can have this full example.

**Review by agnelovaz — CHANGES_REQUESTED** [review]
- I like that the guide is detailed but think each component should first show a simple example so not to scare the user.
- We should be generic in words and examples so not make NUI seem it has limited scope/possible customers.

- Chapters like the drop down look nice with the example images.

- Thinking if some screen shots of applications using the components would make it look even better?

- I resisted grammar comments at this time so not to distract from the main content review.

**Review by adeelkazmi — APPROVED** [review]

**Pull request merged**